tl.lua

```diff
@@ -187,45 +154,86 @@ do
 
       traceback: function(thread, ? string, ? integer): string
       traceback: function(? string, ? integer): string
+      traceback: function(any): any
 
       upvalueid: function(AnyFunction, integer): userdata
       upvaluejoin: function(AnyFunction, integer, AnyFunction, integer)
    end
 
    global record io
       enum OpenMode
-         "r" "w" "a" "r+" "w+" "a+"
-         "rb" "wb" "ab" "r+b" "w+b" "a+b"
-         "*r" "*w" "*a" "*r+" "*w+" "*a+"
+          "r"   "w"   "a"   "r+"   "w+"   "a+"
+          "rb"  "wb"  "ab"  "r+b"  "w+b"  "a+b"
+         "*r"  "*w"  "*a"  "*r+"  "*w+"  "*a+"
          "*rb" "*wb" "*ab" "*r+b" "*w+b" "*a+b"
       end
 
+      enum FileType
+         "file"
+         "closed file"
+      end
+
       close: function(? FILE)
       input: function(? FILE | string): FILE
       flush: function()
 
-      lines: function(? string): (function(): (string))
-      lines: function(? string, FileNumberMode...): (function(): (number...))
-      lines: function(? string, (number | FileStringMode)...): (function(): (string...))
-      lines: function(? string, (number | FileMode)...): (function(): ((string | number)...))
-      lines: function(? string, (number | string)...): (function(): (string...))
+      lines: function(? string): (function(): (string), nil, nil, FILE)
+      lines: function(? string, FileNumberMode...): (function(): (number...), nil, nil, FILE)
+      lines: function(? string, (integer | FileStringMode)...): (function(): (string...), nil, nil, FILE)
```

Comment:
The original reason why all those arguments are `number` rather than `integer` was because the Lua standard library does accept "numbers convertible to integers" in those places, without requiring a `math.tointeger` conversion (in other words, `t = {}; table.insert(t, 1.0, "hello"); print(t[1])` works).

The question is, do we want to reject this in Teal? It would be a breaking change, but we might want to.

Perhaps we should do it in Teal '25?

I'm thinking of cutting one last small 0.24 release, and then merge the `module-split` branch and consider that the start of the '25 series work. (I think I'll branch off a 0.24 maintenance branch just in case some emergency bug comes up after the '25 work picks up steam at the main branch.